plugins/Login/Auth.php

```diff
@@ -60,11 +64,43 @@ public function authenticate()
         return new AuthResult(AuthResult::FAILURE, $this->login, $this->token_auth);
     }
 
-    private function authenticateWithPassword($login, $passwordHash)
+    private function authenticateWithHashedPassword($login, $passwordHash)
     {
         $user = $this->userModel->getUser($login);
 
-        if (!empty($user['login']) && $user['password'] === $passwordHash) {
+        if (!empty($user['login']) && password_verify($passwordHash, $user['password_hashed'])) {
+            return $this->authenticationSuccess($user);
+        }
+
+        return new AuthResult(AuthResult::FAILURE, $login, null);
+    }
+
+    private function authenticateWithPassword($login, $password)
+    {
+        $user = $this->userModel->getUser($login);
+
+        if (empty($user['login'])) {
+            return new AuthResult(AuthResult::FAILURE, $login, null);
+        }
+
+        $legacyOption = sprintf('%s_legacy_password', $user['login']);
```

Comment:
I wonder if we need the option or whether we could always try `password_verify($password, $user['password']) || password_verify(md5($password), $user['password'])` . It may be slightly less secure as it could match when someone enters a wrong password but the chances for this are quite unlikely. Probably it is better to have the option table 👍